tensorflow/stream_executor/stream.h

```diff
@@ -1665,6 +1665,56 @@ class Stream {
                        const DeviceMemory<std::complex<double>> &a, int lda,
                        DeviceMemory<std::complex<double>> *b, int ldb);
 
+  // See BlasSupport::DoBlatLtMatmul.
```

Comment:
ditto - can we simplify these overloadings using the parameter struct I mentioned above? Also, is it easy to make data types / compute types template parameters, or even runtime types? Again, I'd weigh towards simplicity of the signature, as these functions are called for maybe twice throughout Tensorflow.

Reply:
I've now done this with runtime types the same way as in blas.h. I had to add some helper code to work around minor issues that came up with logging and type deduction.

Reply:
I since realized that this wasn't actually calling the template version of DoBlasLtMatmul any more. I've now refactored it again so that it's a template implementation with explicit instantiations.